crates/bevy_ecs/src/system/builder.rs

```diff
@@ -1011,45 +1011,4 @@ mod tests {
             .build_state(&mut world)
             .build_system(|_r: Res<R>, _fr: FilteredResourcesMut| {});
     }
-
-    #[test]
-    #[should_panic]
-    fn filtered_resource_mut_conflicts_write_with_resmut() {
```

Comment:
Why did these tests get removed?  The behavior they're checking seems important to preserve.

Reply:
I think deleting the `filtered_resource_mut_conflicts_write_with_resmut` test was a mistake and I deleted `filtered_resource_reflect` because I had a hard time understanding the test and since `ReflectResource` was changed so much I didn't know how to fix it.

I'll look into this again.